debian/changelog

```diff
@@ -1,8 +1,66 @@
-ubuntu-advantage-tools (1:1+devel) UNRELEASED; urgency=medium
+ubuntu-advantage-tools (32) noble; urgency=medium
 
-  * wip
+  * d/postinst: ensure migrations happen in correct package postinst (GH: #2982)
+  * d/apparmor:introduce new ubuntu_pro_esm_cache apparmor policy
+  * New upstream release 31 (LP: #2060732)
+    - api:
+      + u.pro.attac.token.full_token_attach.v1: add suppport for attach
+        with token
+      + u.pro.services.disable.v1: add support for disable operation
+      + u.pro.services.enable.v1: add support for enable operation
+      + u.pro.detach.v1: add support for detach operation
+      + u.pro.status.is_attached.v1: add extra fields to API response
+      + u.pro.services.dependencies.v1: add support for service dependencies
+      + u.pro.security.fix.*.plan.v1: update ESM cache during plan API
+        if needed
+    - apt_news: add architectures and packages selectors filters for apt news
+    - cli:
+      - improved cli/log message for unexpected errors (GH: #2600)
+      - properly handle setting empty config values (GH: #2925)
+    - cloud-init: support ubuntu_pro user-data
+    - collect-logs: update default output file to pro_logs.tar.gz (LP: #2033313)
+    - config: create public and private config (GH: #2809)
+    - entitlements:
+      + update logic that checks if a service is enabled (LP: #2031192)
+    - fips: warn/confirm with user if enabling fips downgrades the kernel
+    - fix: warn users if ESM cache cannot be updated (GH: #2841)
+    - landscape: use exception for landscape-config failures
```

Comment:
I'm not sure if this one matters much to users - I think its more of an implementation refactor
```suggestion
```